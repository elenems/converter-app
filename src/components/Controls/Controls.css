.MuiFormControl-root{
    width: 16%;
}

.range-filters .MuiFormControl-root{
    width:200px;
}

@media(max-width:991px){
    .inputs .MuiFormControl-root{
        width: 33%;
        margin: 10px 0px;
    }
    .inputs .MuiFormControl-root:nth-child(1){
        order:1;
    }
    .inputs .MuiFormControl-root:nth-child(2){
        order:4;
    }
    .inputs .MuiButton-containedPrimary{
        order:2;
    }
    
    .inputs .MuiFormControl-root:nth-child(5){
        order:3;
    }
    .inputs .MuiFormControl-root:nth-child(4){
        order:5;
    }
    
}