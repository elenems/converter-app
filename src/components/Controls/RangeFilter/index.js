import RangeFilter from './RangeFilter.jsx';
export default RangeFilter;