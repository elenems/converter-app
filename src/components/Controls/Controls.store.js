import Controls from "./Controls.jsx";
import { connect } from "react-redux";
import {
  switchCurrencies,
  handleAmountChange,
  handleCurrencyChange,
  setHistoryRates,
  setHistoryFrom,
  setHistoryTo
} from "../../store/actions/currencyActions";
const mapStateToProps = state => {
  return {
    amountFrom: state.currency.amountFrom,
    amountTo: state.currency.amountTo,
    currencyFrom: state.currency.currencyFrom,
    currencyTo: state.currency.currencyTo,
    historyFrom: state.currency.historyFrom,
    historyTo: state.currency.historyTo,
    error: state.ui.error
  };
};

const mapDispatchToProps = dispatch => {
  return {
    switchCurrencies: () => dispatch(switchCurrencies()),
    handleAmountChange: payload => dispatch(handleAmountChange(payload)),
    handleCurrencyChange: (e, payload) =>
      dispatch(handleCurrencyChange(e, payload)),
    setHistoryRates: payload => dispatch(setHistoryRates(payload)),
    setHistoryFrom: payload => dispatch(setHistoryFrom(payload)),
    setHistoryTo: payload => dispatch(setHistoryTo(payload))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Controls);
