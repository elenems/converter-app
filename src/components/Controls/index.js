import Controls from './Controls.store';
export default Controls;