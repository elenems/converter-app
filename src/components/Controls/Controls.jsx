import React, { Component } from "react";
import "./Controls.css";
import { connect } from "react-redux";
import Box from "@material-ui/core/Box";
import Input from "./Input";
import Select from "./Select";
import Button from "./Button";
import { currencies } from "../../data/currencies";
import RangeFilter from "./RangeFilter";
import {
  switchCurrencies,
  handleAmountChange,
  handleCurrencyChange,
  setHistoryRates,
  setHistoryFrom,
  setHistoryTo
} from "../../store/actions/currencyActions";

const fromCurrencies = currencies;
const toCurrencies = currencies;

class Controls extends Component {
  constructor(props) {
    super(props);
    this.switchCurrecnies = this.switchCurrecnies.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleSelectChange = this.handleSelectChange.bind(this);
    this.setHistoryFrom = this.setHistoryFrom.bind(this);
    this.setHistoryTo = this.setHistoryTo.bind(this);
  }

  switchCurrecnies() {
    const { historyFrom, historyTo, currencyTo, currencyFrom } = this.props;
    this.props.switchCurrencies({ currencyTo, historyTo, historyFrom, currencyFrom });
  }

  handleSelectChange(e) {
    const { historyFrom, historyTo, currencyTo, currencyFrom } = this.props;
    this.props.handleCurrencyChange(e, {
      historyFrom,
      historyTo,
      currencyFrom,
      currencyTo
    });
  }

  handleChange(e) {
    this.props.handleAmountChange(e);
  }

  setHistoryFrom(dateFrom) {
    const {historyTo, currencyTo, currencyFrom } = this.props;
    this.props.setHistoryFrom({
      historyFrom:dateFrom,
      historyTo,
      currencyTo,
      currencyFrom
    });
  }

  setHistoryTo(dateTo) {
    const {historyFrom, currencyTo, currencyFrom } = this.props;
    this.props.setHistoryTo({
      historyFrom,
      historyTo:dateTo,
      currencyTo,
      currencyFrom
    });
  }

  render() {
    const {
      amountFrom,
      currencyFrom,
      currencyTo,
      amountTo,
      historyFrom,
      historyTo
    } = this.props;
    return (
      <div>
        <div className='inputs'>
        <Box
          style={{ margin: "40px 0px 20px" }}
          display="flex"
          flexWrap="wrap"
          justifyContent="space-between"
        >
          <Input
            isDisabled={false}
            inputId="amountFrom"
            value={amountFrom}
            handleChange={this.handleChange}
            fieldLabel="Amount from"
          />

          <Select
            handleChange={this.handleSelectChange}
            selectValue={currencyFrom}
            values={fromCurrencies}
            selectLabel="Currency from"
            selectId="currencyFrom"
          />

          <Button handleClick={this.switchCurrecnies} />

          <Select
            handleChange={this.handleSelectChange}
            selectValue={currencyTo}
            values={toCurrencies}
            selectLabel="Currency to"
            selectId="currencyTo"
          />

          <Input
            isDisabled={true}
            inputId="amountTo"
            value={amountTo}
            handleChange={this.handleChange}
            fieldLabel="Amount To"
          />
        </Box>
        </div>
        <div className='range-filters'>
        <Box className='controls' style={{ margin: "20px 0px" }} display="flex">
          <RangeFilter
            historyChange={this.setHistoryFrom}
            historyRange={historyFrom}
          />
          <RangeFilter historyChange={this.setHistoryTo} historyRange={historyTo} />
        </Box>
        </div>
        {this.props.error ? <p className="error">{this.props.error}</p> : null}
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    amountFrom: state.currency.amountFrom,
    amountTo: state.currency.amountTo,
    currencyFrom: state.currency.currencyFrom,
    currencyTo: state.currency.currencyTo,
    historyFrom: state.currency.historyFrom,
    historyTo: state.currency.historyTo,
    error: state.ui.error
  };
};

const mapDispatchToProps = dispatch => {
  return {
    switchCurrencies: payload => dispatch(switchCurrencies(payload)),
    handleAmountChange: payload => dispatch(handleAmountChange(payload)),
    handleCurrencyChange: (e, payload) =>
      dispatch(handleCurrencyChange(e, payload)),
    setHistoryRates: payload => dispatch(setHistoryRates(payload)),
    setHistoryFrom: payload => dispatch(setHistoryFrom(payload)),
    setHistoryTo: payload => dispatch(setHistoryTo(payload))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Controls);
