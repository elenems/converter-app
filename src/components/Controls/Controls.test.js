import React from "react";
import Controls from "./Controls";
import { shallow } from "enzyme";


describe("Controls", () => {
  it("renders", () => {
    const wrapper = shallow(<Controls />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("handleSelectChange called", () => {
    const mock = jest.fn();
    const wrapper = shallow(<Controls handleCurrencyChange={() => mock()}/>);
    wrapper.instance().handleSelectChange();
    expect(mock).toHaveBeenCalled();
  });

  it("handleChange called", () => {
    const mock = jest.fn();
    const wrapper = shallow(<Controls handleAmountChange={() => mock()}/>);
    wrapper.instance().handleChange();
    expect(mock).toHaveBeenCalled();
  });

  it("setHistoryFrom called", () => {
    const mock = jest.fn();
    const wrapper = shallow(<Controls setHistoryFrom={() => mock()}/>);
    wrapper.instance().setHistoryFrom();
    expect(mock).toHaveBeenCalled();
  });

  it("setHistoryTo called", () => {
    const mock = jest.fn();
    const wrapper = shallow(<Controls setHistoryTo={() => mock()}/>);
    wrapper.instance().setHistoryTo();
    expect(mock).toHaveBeenCalled();
  });

  it("switchCurrecnies called", () => {
    const mock = jest.fn();
    const wrapper = shallow(<Controls switchCurrencies={() => mock()} />);
    wrapper.instance().switchCurrencies();
    expect(mock).toHaveBeenCalled();
  });
});
