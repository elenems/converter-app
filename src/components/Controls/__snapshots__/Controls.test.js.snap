// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls renders 1`] = `
"<div>
  <div className=\\"inputs\\">
    <Styled(MuiBox) style={{...}} display=\\"flex\\" flexWrap=\\"wrap\\" justifyContent=\\"space-between\\">
      <Memo(Input) isDisabled={false} inputId=\\"amountFrom\\" value={[undefined]} handleChange={[Function: bound handleChange]} fieldLabel=\\"Amount from\\" />
      <Memo(Select) handleChange={[Function: bound handleSelectChange]} selectValue={[undefined]} values={{...}} selectLabel=\\"Currency from\\" selectId=\\"currencyFrom\\" />
      <Memo(Button) handleClick={[Function: bound switchCurrencies]} />
      <Memo(Select) handleChange={[Function: bound handleSelectChange]} selectValue={[undefined]} values={{...}} selectLabel=\\"Currency to\\" selectId=\\"currencyTo\\" />
      <Memo(Input) isDisabled={true} inputId=\\"amountTo\\" value={[undefined]} handleChange={[Function: bound handleChange]} fieldLabel=\\"Amount To\\" />
    </Styled(MuiBox)>
  </div>
  <div className=\\"range-filters\\">
    <Styled(MuiBox) className=\\"controls\\" style={{...}} display=\\"flex\\">
      <Memo(RangeFilter) label=\\"History from\\" historyChange={[Function: bound setHistoryFrom]} historyRange={[undefined]} />
      <Memo(RangeFilter) label=\\"History to\\" historyChange={[Function: bound setHistoryTo]} historyRange={[undefined]} />
    </Styled(MuiBox)>
  </div>
</div>"
`;
